### DISCUSSION TITLE
Flashcards app made with customtkinter!

### DISCUSSION THREAD

**Discussion by AngeloM18** [TomSchimansky/CustomTkinter#1760] [Showcase]
Hi, 

![gif](https://user-images.githubusercontent.com/123282394/248436770-a594b86a-a496-49e3-9dce-532fc7d5909f.gif)

I made this app with Customtkinter, Tkinter's treeview class and SQLite for the database. I was wondering if anyone could give the code a look and tell me if there's anything I could have done better. This was my first time using Tkinter and CTK (in general); the documentation for CTK is very well written and concise, so I had no problem getting anything to work.

Anyway, I really enjoyed using CTK, as it looks modern out of the box compared to Tkinter. 

https://github.com/1898Angelo/ankipythia

**Comment by avalon60** [comment]
Have you checked out the row factory technique for returning data from sqlite as dictionaries?

https://intellipaat.com/community/18604/how-can-i-get-dict-from-sqlite-query

Personally, I find it much easier to manage the returned data this way. It's also more robust, in respect of table structure changes.

Also you might try putting docstrings in your classes, methods etc, describing what they are about.

**Reply by AngeloM18** [reply]
Thank you, Avalon.

 - No, I haven't and I really didn't occur to me that I could make queries made with sqlite3 return a dictionary instead, but I'll have a look at it tomorrow night. It seems* super helpful, specially so when you need to handle multiple rows of columnar* data.

 - Yeah, I agree. I need to get used to writing docstrings for classes and methods.
